From 937e4de9f8d36d0daa6b7b2156be90a36ac73dbb Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Mon, 27 Aug 2012 17:03:19 +0800
Subject: [PATCH] lsi/nand:Adjust ECC layout for 4k_4bit pattern

On ACP34xx, remounting a yaffs2 partition will cause some file missing.

This is caused by incorrect ECC layout for 4k_4bit pattern.
The patch adjusts oobfree field of ecc layout for 4k_4bit pattern.

[Extract from vendor drop patch lsi-patch 3.8.1.12.]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/lsi/acp/nand.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/lsi/acp/nand.c b/drivers/lsi/acp/nand.c
index 69b2286..d03ccec 100644
--- a/drivers/lsi/acp/nand.c
+++ b/drivers/lsi/acp/nand.c
@@ -508,7 +508,7 @@ static struct nand_ecclayout lsi_2k_4bit_ecclayout = {
 		40, 41, 42, 43, 44, 45, 46, 47,
 		48, 49, 50, 51, 52, 53, 54, 55,
 		56, 57, 58, 59, 60, 61, 62, 63},
-	.oobfree = {{6,26}}
+	.oobfree = { {2, 30} }
 };
 
 static struct nand_bbt_descr lsi_2k_4bit_bbt_descr = {
-- 
1.7.9.7

